src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -166,11 +176,10 @@ protected internal unsafe override PinValue Read(int pinNumber)
         /// <param name="callback">Delegate that defines the structure for callbacks when a pin value changed event occurs.</param>
         protected internal override void RemoveCallbackForPinValueChangedEvent(int pinNumber, PinChangeEventHandler callback)
         {
-            ValidatePinNumber(pinNumber);
+            ValidatePinIsInput(pinNumber);
```

Comment:
I could see some scenarios where also having callbacks for pins set to output could be useful, for example if the setting of the pin value is happening on a different thread and you want to react to it every time so I wouldn't restrict callbacks only to input pins.

Reply:
See above. Changed to "ValidatePinIsOpen" (ValidatePinNumber just checks that the number is in range)